plugin/pkg/auth/authorizer/rbac/rbac.go

```diff
@@ -121,13 +120,8 @@ func (r *RBACAuthorizer) Authorize(requestAttributes authorizer.Attributes) (boo
 	return false, reason, nil
 }
 
-func New(roles rbacregistryvalidation.RoleGetter, roleBindings rbacregistryvalidation.RoleBindingLister, clusterRoles rbacregistryvalidation.ClusterRoleGetter, clusterRoleBindings rbacregistryvalidation.ClusterRoleBindingLister) *RBACAuthorizer {
-	authorizer := &RBACAuthorizer{
-		authorizationRuleResolver: rbacregistryvalidation.NewDefaultRuleResolver(
-			roles, roleBindings, clusterRoles, clusterRoleBindings,
-		),
-	}
-	return authorizer
+func New(ruleResolver RequestToRuleMapper) *RBACAuthorizer {
```

Comment:
I don't like coercing the interfaces this way.  I do appreciate the difficulty in casting the listers (which you have) to the rbac interfaces (which you don't have), but I'd rather see the shims moved and be exposed here.